tools/cldr-code/src/main/java/org/unicode/cldr/util/ThreadPoolRunner.java

```diff
@@ -0,0 +1,192 @@
+    protected final int start;
+    protected final T context;
+
+    /**
+     * Base class for WriteContexts used with this class.
+     * Encapsulates the parallelism setting.
+     */
+    public static abstract class TPRWriteContext {
+        private int configParallel;
+        private boolean stopped = false;
+
+        public void stop() {
+            stopped = true;
+            logger.info(toString() + " - stopped");
+        }
+
+        public boolean isStopped() {
+            return stopped;
+        }
+
+        public TPRWriteContext() {
+            // setup env
+            CLDRConfig config = CLDRConfig.getInstance();
+
+            // parallelism. 0 means "let Java decide"
+            configParallel = Math.max(config.getProperty("CLDR_THREADPOOLRUNNER_PARALLEL", 0), 0);
+            if (configParallel < 1) {
+                configParallel = java.lang.Runtime.getRuntime().availableProcessors(); // matches ForkJoinPool() behavior
+            }
+            logger.info("CLDR_THREADPOOLRUNNER_PARALLEL=" + configParallel);
+        }
+
+        public int getConfigParallel() {
+            return configParallel;
+        }
+
+        /**
+         * @return number of elements in this context
+         */
+        public abstract int size();
+
+        /**
+         * Default is each item is a chunk.
+         * @return
+         */
+        public int getChunkSize() {
+            return 1;
+        }
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "(" + start + "+" + length + ")";
+    }
+
+    /**
+     * Clone this runner
+     * @param start
+     * @param length
+     * @return
+     */
+    public abstract ThreadPoolRunner<T> clone(int start, int length);
+
+    /**
+     * Constructor for the entirety of the context
+     * @param context
+     */
+    public ThreadPoolRunner(T context) {
+        this(context, 0, context.size());
+    }
+
+    public ThreadPoolRunner(T context, int start, int length) {
+        this.context = context;
+        this.start = start;
+        this.length = length;
+        logger.finer(() -> toString() + " of " + context.size());
+    }
+
+    /**
+     * This is the ForkJoinPool override.
+     * This implementation allows us to be FJP callable
+     */
+    @Override
+    protected void compute() {
+        if (length == 0) {
+            return;
+        } else if (length <= context.getChunkSize()) {
+            computeAll();
+        } else {
+            int split = length / 2;
+            // subdivide
+            invokeAll(clone(start, split),
+                clone(start + split, length - split));
+        }
+    }
+
+    protected void compute(ThreadFactory inFactory) throws InterruptedException {
+        if (length == 0) {
+            return;
+        } else if (length <= context.getChunkSize()) {
+            // below this threshhold, just call directly
+            computeAll();
+        } else {
```

Comment:
this block should be a subroutine, which in turn should call subroutines, so no method is more than about 30 lines

the whole file needs formatting